src/towncrier/test/helpers.py

```diff
@@ -62,10 +63,78 @@ def setup_simple_project(
 ) -> None:
     if config is None:
         config = "[tool.towncrier]\n" 'package = "foo"\n' + extra_config
-
+    else:
+        config = textwrap.dedent(config)
     Path(pyproject_path).write_text(config)
     Path("foo").mkdir()
     Path("foo/__init__.py").write_text('__version__ = "1.2.3"\n')
 
     if mkdir_newsfragments:
         Path("foo/newsfragments").mkdir()
+
+
+def with_project(
+    *,
+    config: str | None = None,
+    pyproject_path: str = "pyproject.toml",
+) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
+    """Decorator to run a test with an isolated directory containing a simple
+    project.
+    Use `config` to tweak the content of the config file.
+    Use `pyproject_path` to tweak the location of the config file.
```

Comment:
Maybe

very minor comment.

```suggestion
    """Decorator to run a test with an isolated directory containing a simple
    project.

    The files are not managed by git.

    `config` is the content of the config file.
    It will be automatically dedented.
    
    `pyproject_path` is the path where to store the config file.
```